src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.ts

```
@@ -32,45 +32,49 @@
-    async generateLearningPath(courseId: number): Promise<void> {
+    async startLearningPath(): Promise<void> {
         try {
-            this.isLearningPathIdLoading.set(true);
-            const learningPathId = await this.learningApiService.generateLearningPath(courseId);
-            this.learningPathId.set(learningPathId);
+            this.isLearningPathLoading.set(true);
+            if (!this.learningPath()) {
+                const learningPath = await this.learningApiService.generateLearningPathForCurrentUser(this.courseId());
+                this.learningPath.set(learningPath);
+            }
+            await this.learningApiService.startLearningPathForCurrentUser(this.learningPath()!.id);
```

Biome

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)